app/controllers/search.controller.js

```diff
@@ -1,55 +1,136 @@
-    data.totalResultsOfKeyword = await page.evaluate(
-      'document.querySelector("#result-stats").innerText'
-    );
-    data.numberOfLinks = await page.$$eval(
-      'a',
-      (as) => as.map((a) => a.href).length
-    );
-    data.totalAdWordsAdvertisers = await page.evaluate(
-      () => document.querySelectorAll('[aria-label="Ads"] > div').length
-    );
-    data.rawHTML = await page.evaluate(
-      () => document.getElementsByTagName('html')[0].innerHTML
-    );
+    try {
+      data.totalResultsOfKeyword = await page.evaluate(
+        'document.querySelector("#result-stats").innerText'
+      );
+      data.numberOfLinks = await page.$$eval(
+        'a',
+        (as) => as.map((a) => a.href).length
+      );
+      data.totalAdWordsAdvertisers = await page.evaluate(
+        () => document.querySelectorAll('[aria-label="Ads"] > div').length
+      );
+      data.rawHTML = await page.evaluate(
+        () => document.getElementsByTagName('html')[0].innerHTML
+      );
+      data.isCompleted = true;
+      Keyword.update(
+        {
+          ...data,
+        },
+        {
+          where: {
+            id: keyword.id,
+          },
+        }
+      );
+    } catch (e) {}
+  }
+
+  browser.close();
+
+  // Run cron too check if the account have in-complete keyword.
+  const task = cron.scheduleJob('*/1 * * * * *', async () => {
+    await crawlKeyword(userId);
+    task.cancel();
+  });
+});
 
-    await Keyword.create({
-      ...data,
+async function crawlKeyword(userId) {
+  Keyword.findAll({
+    where: {
+      userId: userId,
+    },
+    order: [['id', 'DESC']],
+  }).then(async (result) => {
+    if (Boolean(result.length)) {
+      let keyWordUnCrawl = [];
+      for (const keyword of result) {
+        if (!keyword.isCompleted) {
+          keyWordUnCrawl.push(keyword);
+        }
+      }
+      keywordManager.crawl(userId, keyWordUnCrawl);
+    }
+  });
+}
+
+exports.searchByEachKeyWords = async (req, res) => {
+  const { keywords } = req.body;
+
+  if (keywords.length <= 100 && Boolean(keywords.length)) {
+    let arrayValidated = keywords
+      .filter((keyword) => keyword)
+      .map((keyword) => ({
+        keyword: keyword,
+        userId: req.userId,
+        totalResultsOfKeyword: 'Error',
+        numberOfLinks: 0,
+        totalAdWordsAdvertisers: 0,
+        rawHTML: 'Error',
+        isCompleted: false,
+      }));
+    Keyword.bulkCreate(arrayValidated, { individualHooks: true })
+      .then(async function (response) {
+        if (response.length <= 5) {
+          keywordManager.crawl(
+            req.userId,
+            response.slice(0, arrayValidated.length)
+          );
+        } else {
+          keywordManager.crawl(req.userId, response.slice(0, 5));
+        }
+        return res.status(200).send({ message: 'Uploaded' });
+      })
+      .catch(function (error) {
+        res.json(error);
+      });
+  } else {
+    return res.status(500).send({
+      statusCode: 500,
```

Comment:
Server errors (`500`) should be reserved for unhandled errors. Since this is a known error and it is due to incorrect data submitted by the user, the error should be Unprocessable Entity (`422`)

Reply:
Done!